### ISSUE TITLE
linter: eslint/no-useless-constructor bug with access/visibility modifiers

### ISSUE THREAD

**Issue by sebastienbarre** [oxc-project/oxc#15215]
### What version of Oxlint are you using?

1.25.0

### What happened?

Elaborating slightly on the very example found in the documentation: [eslint/no-useless-constructor](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-useless-constructor.html):
```typescript
abstract class A {
  foo: number;
  protected constructor(foo: number) {
    this.foo = foo;
  }
}

class B extends A {
  constructor(foo: number) { // LINT ERROR
    super(foo);
  }
}

const bar = new B(1);
```
is triggering this rule with "Redundant super call in constructor" on class `B`.
The suggestion is "Remove this constructor or add code to it".

This is incorrect though if you can't add code, because you can't remove the *public* constructor in B, since it then inherits the *protected* constructor from A, and you get a TypeScript error when trying to create an instace of B: "Constructor of class 'A' is protected and only accessible within the class declaration.ts(2674)"

```typescript
abstract class A {
  foo: number;
  protected constructor(foo: number) {
    this.foo = foo;
  }
}

class B extends A {
}

const bar = new B(1); // TYPESCRIPT ERROR
```

**Comment by camc314** [comment]
I've added a "caveat" section to the rule documentation.

This rule is based on the typescript-eslint rule, hence we're following their behaviour https://github.com/typescript-eslint/typescript-eslint/issues/3820#issuecomment-917821240

**Comment by sebastienbarre** [comment]
I’m a bit confused by this:
* I’m running in `typeAware` mode, which was introduced back in August 2025; why is type information missing?
* Could you clarify why this rule is categorized under the `typescript/ namespace` if it checks a purely JavaScript feature (class constructors) but still causes TypeScript compiler errors?
* The 2021 comment you linked seems outdated; could the correct behavior be implemented behind an option instead? This issue produces a reproducible false positive.

Thank you for your time and clarification.
